internal/backend/runtime/omni/controllers/omni/internal/task/machine/poll.go

```diff
@@ -99,6 +99,9 @@ func pollVersion(ctx context.Context, c *client.Client, info *Info) error {
 }
 
 func pollHostname(ctx context.Context, c *client.Client, info *Info) error {
+	info.Hostname = pointer.To("")
```

Comment:
this works, but just fyi: the way Talos works, on boot, there might be a window when hostname is not known, and it might flicker a bit between `""` and whatever known hostname.

not a big deal though

Reply:
I think if the machine reboots it's machine status gets updated anyways.